server/src/test/java/org/elasticsearch/indices/cluster/ClusterStateChanges.java

```diff
@@ -210,7 +217,35 @@ public ClusterState createIndex(ClusterState state, CreateIndexRequest request)
     }
 
     public ClusterState closeIndices(ClusterState state, CloseIndexRequest request) {
-        return execute(transportCloseIndexAction, request, state);
+        final ClusterState[] result = new ClusterState[1];
+        doAnswer(invocation -> {
+            // Closing indices is a multiple steps process that updates the cluster state, reaches to the shards to execute verifications
+            // and finally updates the cluster state again. This cannot be easily reproduced here, so we'll execute the first cluster
+            // update and then simulates that the verification succeed on shards by manually updating the cluster state again.
```

Comment:
I think this can be much simplified by just mocking out the second step (see my earlier comment)